apis/alloydb/v1alpha1/instance_types.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var AlloyDBInstanceGVK = GroupVersion.WithKind("AlloyDBInstance")
+
+// AlloyDBInstanceSpec defines the desired state of AlloyDBInstance
+// +kcc:proto=google.cloud.alloydb.v1beta.Instance
+type AlloyDBInstanceSpec struct {
+
+	// The AlloyDBInstance cluster that this resource belongs to.
+	// +required
+	ClusterRef *refs.AlloyDBClusterRef `json:"clusterRef,omitempty"`
```

Comment:
(Just notice this, maybe we should start from AlloyDBCluster first at the beginning)

Reply:
Ack.